### ISSUE TITLE
Hot Reload breaking on rename file (and change the path)

### ISSUE THREAD

**Issue by TutorFx** [nuxt/nuxt#25631]
### Environment

- Operating System: Windows_NT
- Node Version:     v21.6.1
- Nuxt Version:     3.10.0
- CLI Version:      3.10.0
- Nitro Version:    2.8.1
- Package Manager:  pnpm@8.13.1
- Builder:          -
- User Config:      site, app, devtools, srcDir, css, modules, image, vite, eslint
- Runtime Modules:  nuxt-site-config@2.2.9, nuxt-speedkit@3.0.0-next.45, @nuxtseo/module@2.0.0-beta.55,
 @nuxtjs/tailwindcss@6.11.3, @nuxtjs/eslint-module@4.1.0, @nuxt/image@1.3.0, nuxt-icon@0.6.8
- Build Modules:    -

### Reproduction

It's not reproductible with a repository

### Describe the bug

I was able to verify that by changing the name of a component and changing its path so that this file can be imported in the same way, the hot reload is broken.

before:
src\components\secondary\ExampleComponent.vue
after:
src\components\secondary\Example\Component.vue

### Additional context

I achieved this result by renaming the file through VSC.

### Logs

```shell-script
Pre-transform error: Failed to load url /components/secondary/ExampleComponent.vue (resolved id: C:/Users/<censored>/Documents/GitHub/project-name/src/components/secondary/ExampleComponent.vue) in C:/Users/<censored>/Documents/GitHub/project-name/src/components/secondary/Example.vue. Does the file exist?
```

**Comment by KnightYoshi** [comment]
Running into this on macOS as well. Nuxt is running inside of a Docker container. Added files are detected, but existing files removed, renamed, or moved break the HMR and I have to restart the whole Nuxt app

**Comment by danielroe** [comment]
This should have been resolved in v3.14.1592 via https://github.com/nuxt/nuxt/pull/29954.
